src/models/Scholarships.ts

```diff
@@ -100,6 +98,13 @@ class Scholarships extends FirestoreCollection<ScholarshipData> {
 
     return FirestoreCollection.list(query, postProcessFilter);
   }
+
+  save(data: ScholarshipData): Promise<Model<ScholarshipData>> {
+    const Data = { ...data, lastModified: new Date() };
+    if (!data.dateAdded) Data.dateAdded = new Date();
```

Comment:
There's a slight time difference between the two lines. I suggest doing this instead so that the date values are consistent for new scholarships:
```suggestion
    if (!data.dateAdded) newData.dateAdded = newData.lastModified;
```